backend/src/app/storage/gc_touched.clj

```diff
@@ -0,0 +1,208 @@
+  maybe this storage_object is no longer necessary and can be eligible
+  for elimination. This task periodically analyzes touched objects and
+  mark them as freeze (means that has other references and the object
+  is still valid) or deleted (no more references to this object so is
+  ready to be deleted)."
+  (:require
+   [app.common.data :as d]
+   [app.common.data.macros :as dm]
+   [app.common.exceptions :as ex]
+   [app.common.logging :as l]
+   [app.db :as db]
+   [app.storage :as-alias sto]
+   [app.storage.impl :as impl]
+   [clojure.spec.alpha :as s]
+   [integrant.core :as ig]))
+
+(def ^:private sql:get-team-font-variant-nrefs
+  "SELECT ((SELECT count(*) FROM team_font_variant WHERE woff1_file_id = ?) +
+           (SELECT count(*) FROM team_font_variant WHERE woff2_file_id = ?) +
+           (SELECT count(*) FROM team_font_variant WHERE otf_file_id = ?) +
+           (SELECT count(*) FROM team_font_variant WHERE ttf_file_id = ?)) AS nrefs")
+
+(defn- get-team-font-variant-nrefs
+  [conn id]
+  (-> (db/exec-one! conn [sql:get-team-font-variant-nrefs id id id id])
+      (get :nrefs)))
+
+
+(def ^:private
+  sql:get-file-media-object-nrefs
+  "SELECT ((SELECT count(*) FROM file_media_object WHERE media_id = ?) +
+           (SELECT count(*) FROM file_media_object WHERE thumbnail_id = ?)) AS nrefs")
+
+(defn- get-file-media-object-nrefs
+  [conn id]
+  (-> (db/exec-one! conn [sql:get-file-media-object-nrefs id id])
+      (get :nrefs)))
+
+
+(def ^:private sql:get-profile-nrefs
+  "SELECT ((SELECT count(*) FROM profile WHERE photo_id = ?) +
+           (SELECT count(*) FROM team WHERE photo_id = ?)) AS nrefs")
+
+(defn- get-profile-nrefs
+  [conn id]
+  (-> (db/exec-one! conn [sql:get-profile-nrefs id id])
+      (get :nrefs)))
+
+
+(def ^:private
+  sql:get-file-object-thumbnail-nrefs
+  "SELECT (SELECT count(*) FROM file_tagged_object_thumbnail WHERE media_id = ?) AS nrefs")
+
+(defn- get-file-object-thumbnails
+  [conn id]
+  (-> (db/exec-one! conn [sql:get-file-object-thumbnail-nrefs id])
+      (get :nrefs)))
+
+
+(def ^:private
+  sql:get-file-thumbnail-nrefs
+  "SELECT (SELECT count(*) FROM file_thumbnail WHERE media_id = ?) AS nrefs")
+
+(defn- get-file-thumbnails
+  [conn id]
+  (-> (db/exec-one! conn [sql:get-file-thumbnail-nrefs id])
+      (get :nrefs)))
+
+
+(def ^:private sql:mark-freeze-in-bulk
+  "UPDATE storage_object
+      SET touched_at = NULL
+    WHERE id = ANY(?::uuid[])")
+
+(defn- mark-freeze-in-bulk!
+  [conn ids]
+  (let [ids (db/create-array conn "uuid" ids)]
+    (db/exec-one! conn [sql:mark-freeze-in-bulk ids])))
+
+
+(def ^:private sql:mark-delete-in-bulk
+  "UPDATE storage_object
+      SET deleted_at = now(),
+          touched_at = NULL
+    WHERE id = ANY(?::uuid[])")
+
+(defn- mark-delete-in-bulk!
+  [conn ids]
+  (let [ids (db/create-array conn "uuid" ids)]
+    (db/exec-one! conn [sql:mark-delete-in-bulk ids])))
+
+;; NOTE: A getter that retrieves the key witch will be used
```

Comment:
witch == "a person thought to have magic powers, especially evil ones" , wich? :D